libgalois/include/katana/PropertyManager.h

```diff
@@ -19,11 +19,15 @@ class KATANA_EXPORT PropertyManager : public Manager {
   const std::string& MemoryCategory() const override { return memory_category; }
   count_t FreeStandbyMemory(count_t goal) override;
 
-  void AddProperty(const katana::Uri& property_path);
+  /// Client wants a property, see if we have it in the cache and if so return it and make the memory active.
+  /// Returns nullptr if manager does not have it in the cache
+  std::shared_ptr<arrow::Table> GetProperty(const katana::Uri& property_path);
```

Comment:
Does this mean we have settled on file name as the one true cache key?

Reply:
I was very excited by this prospect, as the random tag in the file name is the right level of indirection for the cache instead of rdg_dir+version+view.  But there are some subtleties to caching dirty properties that will probably complicate this glorious vision.